### PR TITLE
TT-39 feat: allow switching amongyears

### PR DESCRIPTION
**Pull request by scastillo-jp**

### CODE REVIEW

File: src/app/modules/shared/components/month-picker/month-picker.component.html [hunk 1]
```diff
@@ -1,10 +1,59 @@
...
+  -  <div
+  -    class="month w-100 text-center align-self-center d-flex flex-column"
+  -    *ngFor="let month of months; let i = index"
+  -    (click)="getMonth(i)"
+  -    [ngClass]="{ active: activeMonth === i }"
+  -  >
+  -   <div class="p-2">{{ month }}</div>
+  -  </div>
+  -</div>
+   -->
```
Comment (enriquezrene) [inline src/app/modules/shared/components/month-picker/month-picker.component.html:11]: should not you remove this dead code?

File: src/app/modules/shared/components/month-picker/month-picker.component.ts [hunk 2]
```diff
@@ -1,23 +1,133 @@
...
+    this.model.selectMonth(index);
+    this.onChange(this.model.selectedMonthIndex, this.model.selectedMonthYear);
+  }
+
+  isSelectedMonth(monthIndex: number) {
+    return this.model.selectedMonthIndex === monthIndex && this.model.selectedMonthYear === this.model.selectedYearMoment.year();
+  }
+
+  onChange(monthIndex: number, year: number) {
+    // tslint:disable-next-line:no-unused-expression
```
Comment (enriquezrene) [inline src/app/modules/shared/components/month-picker/month-picker.component.ts:60]: no reason to have this line here.

File: src/app/modules/shared/components/month-picker/month-picker.component.ts [hunk 3]
```diff
@@ -1,23 +1,133 @@
...
+    this.changeDate.emit({monthIndex, year});
+  }
+
+  // tslint:disable-next-line:no-unused-expression
```
Comment (enriquezrene) [inline src/app/modules/shared/components/month-picker/month-picker.component.ts:64]: no reason to have this line here.

File: src/app/modules/shared/components/month-picker/month-picker.component.ts [hunk 4]
```diff
@@ -1,23 +1,133 @@
...
+  }
+
+  renderYears() {
+    this.years = [];
+    for (let i = 5; i > 0; i--) {
+      this.years.push(this.model.selectedYearMoment.year() - i);
+    }
+    for (let i = 0; i <= 6; i++) {
+      this.years.push(this.model.selectedYearMoment.year() + i);
+    }
```
Comment (enriquezrene) [inline src/app/modules/shared/components/month-picker/month-picker.component.ts:77]: What are you trying to do here? is this better? 
```    
    const currentYear = this.model.selectedYearMoment.year();
    for (let i = currentYear -5; i < currentYear+5; i++) {
      if( i !== currentYear){ this.years.push(i); }
    }
```

File: src/app/modules/shared/components/month-picker/month-picker.component.ts [hunk 5]
```diff
@@ -1,23 +1,133 @@
...
+  }
+
+  selectYear(year: number) {
+    this.isShowYears = false;
+    this.model.selectedYearMoment = moment().year(year);
+    this.model.updateYearText(); // in set get aendern
```
Comment (enriquezrene) [inline src/app/modules/shared/components/month-picker/month-picker.component.ts:83]: What do you mean?
```
// in set get aendern
```

File: src/app/modules/time-entries/pages/time-entries.component.ts [hunk 6]
```diff
@@ -36,7 +40,7 @@ export class TimeEntriesComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit(): void {
-    this.store.dispatch(new entryActions.LoadEntries(new Date().getMonth() + 1));
+    // this.store.dispatch(new entryActions.LoadEntries(new Date().getMonth() + 1));
```
Comment (enriquezrene) [inline src/app/modules/time-entries/pages/time-entries.component.ts:43]: can we remove this commented line?

File: src/app/modules/time-entries/pages/time-entries.component.ts [hunk 7]
```diff
@@ -160,13 +164,21 @@ export class TimeEntriesComponent implements OnInit, OnDestroy {
...
 
   openModal(item: any) {
     this.idToDelete = item.id;
     this.message = `Are you sure you want to delete ${item.activity_name}?`;
     this.showModal = true;
   }
+
+  // getMonth(month: number) {
+  //   this.store.dispatch(new entryActions.LoadEntries(month));
+  // }
```
Comment (enriquezrene) [inline src/app/modules/time-entries/pages/time-entries.component.ts:176]: can we remove this commented code?

**Review by enriquezrene — CHANGES_REQUESTED** [review]

File: src/app/modules/shared/components/month-picker/month-picker.component.ts [hunk 8]
```diff
@@ -1,23 +1,133 @@
...
+
+  // tslint:disable-next-line:no-unused-expression
+  toggleShowYears() {
+    this.isShowYears = !this.isShowYears;
+    this.renderYears();
+  }
+
+  renderYears() {
+    this.years = [];
+    for (let i = 5; i > 0; i--) {
```
Comment (juanultimate) [inline src/app/modules/shared/components/month-picker/month-picker.component.ts:72]: What are these [magic numbers](https://refactoring.guru/es/replace-magic-number-with-symbolic-constant) doing here?

**Pull request closed without merging**